### PR TITLE
Add iOS targets

### PR DESCRIPTION
**Pull request by Vichy97**
Add support for iOS targets

### CODE REVIEW

File: gradle/build-logic/convention/src/main/kotlin/KotlinMultiplatformConventionPlugin.kt [hunk 1]
```diff
@@ -23,6 +23,9 @@ class KotlinMultiplatformConventionPlugin : Plugin<Project> {
     extensions.configure<KotlinMultiplatformExtension> {
       applyDefaultHierarchyTemplate()
       jvm("desktop")
+      iosArm64()
+      iosX64()
+      iosSimulatorArm64()
```
Comment (Vichy97) [inline gradle/build-logic/convention/src/main/kotlin/KotlinMultiplatformConventionPlugin.kt:28]: Not familiar with publishing KMP libraries for iOS. I think we may need the equivalent to `publishLibraryVariants` but for iOS?

Comment (saket) [reply gradle/build-logic/convention/src/main/kotlin/KotlinMultiplatformConventionPlugin.kt:28]: Yea I'm not sure either. Let's try merging this and seeing if a SNAPSHOT version gets published correctly for iOS.

**Review by saket — APPROVED** [review]

File: gradle/build-logic/convention/src/main/kotlin/KotlinMultiplatformConventionPlugin.kt (hunk 1 above)
Comment (Vichy97) [reply gradle/build-logic/convention/src/main/kotlin/KotlinMultiplatformConventionPlugin.kt:28]: Any ideas on the failures on this PR? The error mentions a missing android target, but I did not modify that 🤔

Comment (saket) [reply gradle/build-logic/convention/src/main/kotlin/KotlinMultiplatformConventionPlugin.kt:28]: Sorry I'm just seeing this. Let me try building the project on my machine.

**Comment by wang222222222** [comment]
Hi any update here ?

**Comment by saket** [comment]
Closing this PR as I ended up [merging the commit](https://github.com/saket/telephoto/commit/cb193fbd3d058cfb437f786c8881c495a88cddf3) manually. Git wasn't letting me force-push to your branch @Vichy97.

**Comment by saket** [comment]
@wang222222222 @Vichy97 iOS targets are now available with `0.15.0-SNAPSHOT`. Would either of you be able to try it out?

**Comment by Vichy97** [comment]
@saket https://mvnrepository.com/artifact/me.saket.telephoto/zoomable I don't the snapshot version on Maven.

**Comment by saket** [comment]
Snapshot builds are available in the snapshot repository:

```gradle
maven("https://oss.sonatype.org/content/repositories/snapshots/")
```

**Comment by Vichy97** [comment]
Hmm it doesn't appear to have ios artifacts. was able to use it in Android using the snapshot though.

**Comment by Vichy97** [comment]
I think you maybe need more changes than just adding the iOS target that I had in my pr, but I'm honestly not sure what else you need for the iOS targets to be uploaded.

**Comment by saket** [comment]
Interesting. It looks like telephoto will need to enable this option: https://kotlinlang.org/docs/multiplatform-publish-lib.html#host-requirements

**Comment by Vichy97** [comment]
Ah that makes sense. I'm happy to try the next snapshot if you add that change! I could also try to make a PR soon

**Comment by saket** [comment]
@Vichy97 can you give it another try? https://oss.sonatype.org/content/repositories/snapshots/me/saket/telephoto/zoomable-iosarm64/0.15.0-SNAPSHOT/

cc @vanniktech

**Comment by vanniktech** [comment]
Yup works for me on iOS.

**Comment by vanniktech** [comment]
Weirdly when I change:

```diff
-telephoto = "me.saket.telephoto:zoomable:0.15.0-SNAPSHOT"
+telephoto = "me.saket.telephoto:zoomable-image-coil3:0.15.0-SNAPSHOT"
```

Android Studio sync throws up:

> org.gradle.api.internal.artifacts.ivyservice.TypedResolveException: Could not resolve all files for configuration ':composeApp:iosArm64CompilationDependenciesMetadata'.

So the zoomable module works on iOS, the zoomable-image-coil3, does build but Gradle Sync is broken. Is the iosArm64 variant missing there?

**Comment by saket** [comment]
Yep, `SubSamplingImage()` needs [more work](https://github.com/saket/telephoto/issues/9#issue-1709426327) to make it multiplatform :(

**Comment by Vichy97** [comment]
@saket everything seems to be working now. Thanks for doing that!

**Comment by vanniktech** [comment]
Is `SubSamplingImage` required for `ZoomableAsyncImage`? Or would it be possible to split it up in two modules so that `ZoomableAsyncImage` can also be used on JVM.

**Comment by saket** [comment]
Yea this should be doable now. Since Coil 3.x restricts the maximum size of bitmaps, large images will not cause a crash even if they can't be sub-sampled. Time to start moving `zoomable-image:core` to shared code.

**Comment by vanniktech** [comment]
I've ended up using the `Modifier.zoomable()` so I'm all good for now with the SNAPSHOT.

**Pull request closed without merging**